### PR TITLE
Update  CHANGELOG.md for 7.3.0

### PR DESCRIPTION
**Pull request by tbradellis**
Updates change log with release notes for 7.3.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Coming soon
 * TBD
 
+## Version 7.3.0 (2021-9-30)
+
+### New features and improvements
+
+* Support for [Java 16](https://github.com/newrelic/newrelic-java-agent/pull/366)
```
Comment (Stephan202) [inline CHANGELOG.md:14]: :pray: Do you think #433 will make it into the release as well?

(I tested #366 + #437 + the current state of #433 on top of release 7.2.0, i.c.w. a JDK 17 base image, and it seemed to work fine. Class files did still target Java 11, though.)

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+* Auto-instrumentation support for [java.net.http.HttpClient](https://github.com/newrelic/newrelic-java-agent/pull/251) 
+
+* Migrate the Agent’s caching library from Guava to [Caffeine](https://github.com/newrelic/newrelic-java-agent/pull/295)
+  * [Caffeine provides an in-memory cache](https://github.com/ben-manes/caffeine#cache) using a Google Guava inspired API. The improvements draw on the author’s experience designing Guava's cache and ConcurrentLinkedHashMap.
```
Comment (Stephan202) [inline CHANGELOG.md:19]:
```suggestion
  * [Caffeine provides an in-memory cache](https://github.com/ben-manes/caffeine#cache) using a Google Guava inspired API. The improvements draw on the author’s experience designing Guava's cache and `ConcurrentLinkedHashMap`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  * This change will provide improvements in cases where we saw thread contention and deadlocks attributable to the Guava library
```
Comment (Stephan202) [inline CHANGELOG.md:20]:
```suggestion
  * This change will provide improvements in cases where we saw thread contention and deadlocks attributable to the Guava library.
```

("will" or "is expected to" :thinking:)

Comment (tbradellis) [reply CHANGELOG.md:20]: Good call. "Is expected" would be more appropriate here.

File: CHANGELOG.md (hunk 1 above)
Comment (tbradellis) [reply CHANGELOG.md:14]: Doesn't look like we'll be able to get #433 in 7.3.0.  Our current testing is looking good as well, but we have some testing goals and reviews to hit that would make it a rush job to get finished before 7.3.0 releases. We expect 17 support to be pretty quick on the heels of that release though.

**Review by jasonjkeller — APPROVED** [review]
Good to 🐐

**Pull request merged**